status-core/src/main/java/com/indeed/status/core/AbstractDependencyManager.java

```diff
@@ -327,6 +336,11 @@ public void addDependency(final Dependency dependency) {
     }
 
     public Dependency removeDependency(final String id) {
+        if (dependencyPingers.containsKey(id)) {
+            // Cancel all future pings for this dependency, interrupting any current pings
+            dependencyPingers.remove(id).cancel(true);
```

Comment:
I think we might need to make sure `dependencyPingers.remove(id)` doesn't return null.

Reply:
updated